qcoro/qcorotask.h

```diff
@@ -742,23 +772,40 @@ inline T waitFor(QCoro::Task<T> &&task) {
  *        For void tasks, it needs to have no arguments.
  *        For all other types, it takes a value of the type as single argument.
  */
-template <typename T, typename Callback>
-requires std::is_invocable_v<Callback> || std::is_invocable_v<Callback, T>
-void connect(QCoro::Task<T> &task, QObject *context, Callback func) {
+template <typename T, typename QObjectSubclass, typename Callback>
+requires std::is_invocable_v<Callback> || std::is_invocable_v<Callback, T> || std::is_invocable_v<Callback, QObjectSubclass>
+void connect(QCoro::Task<T> &&task, QObjectSubclass *context, Callback func) {
     QPointer ctxWatcher = context;
     if constexpr (std::is_same_v<T, void>) {
-        task.then([ctxWatcher, func = std::move(func)]() {
+        task.then([ctxWatcher, func = std::move(func), context]() {
```

Comment:
There's no need to capture `context`, you already have the same object captured via `ctxWatcher`.